company-coq.el

```diff
@@ -2218,14 +2217,39 @@ in that file."
   "Find qualified name of NAME using CMD-FORMAT and RESPONSE-HEADERS."
   (company-coq--fqn-with-regexp-1 name cmd-format (company-coq--loc-output-regexp response-headers)))
 
+(defun company-coq--loc-from-globfile (globfile glob_relative_fqn)
+  "Find qualified char offset of GLOB_RELATIVE_FQN in GLOBFILE. GLOB_RELATIVE_FQN the fully qualified name (fqn) relative to the fqn of the file, except that it is in the globfile format: the last . is replaced by a space"
+  (message "looking for %s in globfile %s" glob_relative_fqn globfile)
+  (-when-let* ((globfile-content (f-read-text globfile))
+	       (startindex  (save-match-data
+			      (and (string-match (company-coq-make-headers-regexp-glob glob_relative_fqn) globfile-content)
+				   (match-string-no-properties 2 globfile-content)))))
+    (string-to-number startindex)))
+
+(defun glob_relative_fqn (vfile_fqn short-name fqn)
+  "compute the name of this definition as expected to be in the .glob file in the line corresponding to the declartion side. fqn is the fully qualified name (fqn) of this definition. vfile_fqn is the fqn of the .v file corresponding to the .glob file. shortname is the leaf of fqn"
+  (if (string= (concat vfile_fqn "." short-name) fqn)
+      (concat "<> " short-name)
+      (let ((relative_fqn_except_leaf (substring fqn (1+(length vfile_fqn)) (-(1+(length short-name))))));; the middle part of fqn excluding the fully qualified name for the file and excluding the leaf
+	(concat relative_fqn_except_leaf " " short-name))))
+
 (defun company-coq--loc-with-regexp (name cmd-format response-headers)
   "Find location of NAME using CMD-FORMAT and RESPONSE-HEADERS.
 Returns a cons as specified by `company-coq--locate-name'."
   (-when-let* ((fqn (company-coq--fqn-with-regexp name cmd-format response-headers))
-               (loc (company-coq--loc-fully-qualified-name fqn))
-               (short-name (replace-regexp-in-string "\\`.*\\." "" fqn)))
-    (cons loc (concat (company-coq-make-headers-regexp response-headers)
-                      "\\s-*\\(" (regexp-quote short-name) "\\)\\_>"))))
+	 (short-name (replace-regexp-in-string "\\`.*\\." "" fqn)))
+  (or (-when-let* ((loc (company-coq--loc-fully-qualified-name fqn));; will fail iff the definition is in current file
+	       (vfile (concat (replace-regexp-in-string "_build/default" "" (car loc) nil 'literal) ".v"))
```

Comment:
to be on the safer side, check that the parent directory of `_build` has a `dune-project` file